media-sound/teamspeak-server/teamspeak-server-3.3.0.ebuild

```diff
@@ -0,0 +1,116 @@
+
+DESCRIPTION="A server software for hosting quality voice communication via the internet"
+HOMEPAGE="https://www.teamspeak.com/"
+SRC_URI="amd64? ( http://ftp.4players.de/pub/hosted/ts3/releases/${PV}/teamspeak3-server_linux_amd64-${PV}.tar.bz2 )
+	x86? ( http://ftp.4players.de/pub/hosted/ts3/releases/${PV}/teamspeak3-server_linux_x86-${PV}.tar.bz2 )"
+
+LICENSE="LGPL-2.1 teamspeak3"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+IUSE="doc mysql tsdns"
+
+RESTRICT="mirror strip"
+
+S="${WORKDIR}/teamspeak3-server_linux"
+
+QA_PREBUILT="opt/teamspeak3-server/libmariadb.so.2
+	opt/teamspeak3-server/libts3db_mariadb.so
+	opt/teamspeak3-server/libts3db_sqlite3.so
+	opt/teamspeak3-server/libts3_ssh.so
+	opt/teamspeak3-server/ts3server
+	opt/teamspeak3-server/tsdnsserver"
+
+pkg_setup() {
+	enewgroup teamspeak
+	enewuser teamspeak -1 -1 /opt/teamspeak3-server teamspeak
+}
+
+src_unpack() {
+	unpack ${A}
+
+	mv teamspeak3-server_linux_$(usex amd64 amd64 x86) teamspeak3-server_linux || die
+}
+
+src_install() {
+	touch "${T%/}"/.ts3server_license_accepted || die
+	insinto "/opt/teamspeak3-server"
+	doins "${T}"/.ts3server_license_accepted
+
+	keepdir /etc/teamspeak3-server
+	diropts -o teamspeak -g teamspeak
+	keepdir /opt/teamspeak3-server /var/log/teamspeak3-server
+
+	exeinto /opt/teamspeak3-server
+	doexe ts3server
+	doexe "${FILESDIR}"/ts3server-bin
+	dodir /opt/bin
+	dosym ../teamspeak3-server/ts3server-bin /opt/bin/ts3server
+
+	exeinto /opt/teamspeak3-server
+	doexe libts3db_sqlite3.so libts3_ssh.so
+
+	docinto	serverquery
+	dodoc serverquerydocs/*.txt
+
+	dosym ../../usr/share/doc/teamspeak-server-${PV}/serverquery /opt/teamspeak3-server/serverquerydocs
+
+	insinto /opt/teamspeak3-server/sql
+	doins sql/*.sql
+	doins -r sql/create_sqlite
+
+	insinto /etc/teamspeak3-server
+	newins "${FILESDIR}"/ts3server.ini-r1 ts3server.ini
+
+	newinitd "${FILESDIR}"/teamspeak.initd-r1 teamspeak3-server
+	systemd_newunit "${FILESDIR}"/teamspeak.service-r1 teamspeak3-server.service
+
+	if use mysql; then
+		insinto /etc/teamspeak3-server
+		newins "${FILESDIR}"/ts3server_mariadb.ini.sample-r1 ts3server_mariadb.ini.sample
+		doins "${FILESDIR}"/ts3db_mariadb.ini.sample
+
+		exeinto /opt/teamspeak3-server
+		doexe libts3db_mariadb.so
+		doexe redist/libmariadb.so.2
+
+		insinto /opt/teamspeak3-server/sql
+		doins -r sql/create_mariadb
+		doins -r sql/updates_and_fixes
+	fi
+
+	if use doc; then
+		docinto serverquery/html
+		dodoc doc/serverquery/*
+	fi
+
+	if use tsdns; then
+		exeinto /opt/teamspeak3-server
+		doexe tsdns/tsdnsserver
+		dodir /opt/bin
+		dosym ../teamspeak3-server/tsdnsserver /opt/bin/tsdnsserver
+
+		insinto /etc/teamspeak3-server
+		doins tsdns/tsdns_settings.ini.sample
+
+		docinto ts3server
+		dodoc doc/*.txt
+
+		docinto tsdns
+		dodoc tsdns/README tsdns/USAGE
+	fi
+
+	echo "CONFIG_PROTECT=\"/etc/teamspeak3-server/ts3server.ini /etc/teamspeak3-server/ts3server_mariadb.ini\"" > "${T}"/99teamspeak3-server || die
+	doenvd "${T}"/99teamspeak3-server
```

Comment:
You could use `newenvd` with stdin input here as well, and `<<<` or heredoc to make it more readable.

Reply:
They fail with pypy for me, so removed.
-> https://pastebin.com/8Par3Ke0

Reply:
Done.

Reply:
That's not what I said. 
```
newenvd - foo <<-EOF || die
...